bundles/org.openhab.binding.teleinfo/src/main/java/org/openhab/binding/teleinfo/internal/reader/io/TeleinfoInputStream.java

```diff
@@ -164,137 +78,65 @@ public void close() throws IOException {
      * @throws InvalidFrameException
      * @throws Exception
      */
-    public synchronized @Nullable Frame readNextFrame() throws TimeoutException, InvalidFrameException, IOException {
+    public synchronized @Nullable Frame readNextFrame() throws InvalidFrameException, IOException {
         // seek the next header frame
-        Future<@Nullable Void> seekNextHeaderFrameTask = executorService.submit(() -> {
-            while (!isHeaderFrame(groupLine)) {
-                groupLine = bufferedReader.readLine();
-                if (logger.isTraceEnabled()) {
-                    logger.trace("groupLine = {}", groupLine);
-                }
-                if (groupLine == null) { // end of stream
-                    logger.trace("end of stream reached !");
-                    return null;
-                }
+        while (!isHeaderFrame(groupLine)) {
+            groupLine = bufferedReader.readLine();
```

Comment:
So you aren't handling timeouts anymore?

Reply:
This was suggested by @fwolter during the review of our initial contribution but we didn't follow his advice.
Since then I discovered the timeouts make the binding quite unusable with rfc2217 serial port: I have to pause and resume the bridge several before getting online.
Do you think there can be some drawbacks without timeouts?